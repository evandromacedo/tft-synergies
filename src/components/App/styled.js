import { createGlobalStyle } from 'styled-components';
import BeaufortRegular from '../../static/font/Beaufort-Regular.ttf';
import BeaufortBold from '../../static/font/Beaufort-Bold.ttf';

export const GlobalStyle = createGlobalStyle`
  /* Resets */
  *, *:before, *:after {
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }

  html, body, div, span, applet, object, iframe,
  h1, h2, h3, h4, h5, h6, p, blockquote, pre,
  a, abbr, acronym, address, big, cite, code,
  del, dfn, em, img, ins, kbd, q, s, samp,
  small, strike, strong, sub, sup, tt, var,
  b, u, i, center,
  dl, dt, dd, ol, ul, li,
  fieldset, form, label, legend,
  table, caption, tbody, tfoot, thead, tr, th, td,
  article, aside, canvas, details, embed,
  figure, figcaption, footer, header, hgroup,
  menu, nav, output, ruby, section, summary,
  time, mark, audio, video {
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
  }

  body {
    -webkit-tap-highlight-color: transparent;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    line-height: 1;
  }

  ol, ul {
    list-style: none;
  }

  a {
    color: inherit;
    text-decoration: inherit;
  }

  input {
    outline-width: 0.3em;
    outline-color: transparent;
    -webkit-appearance: none;
    appearance: none;
    -webkit-border-radius:0;
    border-radius:0;
  }

  .input:disabled {
    background-color: transparent;
  }

  button {
    outline: none;
    cursor: pointer;
    background-color: transparent;
    margin: 0;
  }

  input,
  button {
    text-transform: inherit;
    font-family: inherit;
    letter-spacing: inherit;
    border: none;
  }

  /* Responsiveness */
  img,
  iframe,
  object,
  embed,
  video {
    height: auto;
    max-width: 100%;
  }

  /* Fonts */
  @font-face {
    font-family: 'Beaufort';
    src: url(${BeaufortRegular}) format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
    font-family: 'Beaufort';
    src: url(${BeaufortBold}) format('truetype');
    font-weight: bold;
    font-style: normal;
  }

  html {
    font-family: 'Beaufort', sans-serif;
    font-weight: bold;
    font-size: 10px;
    color: #fff;
    background-color: #121212;
  }
`;
