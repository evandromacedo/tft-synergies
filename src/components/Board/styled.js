import styled from 'styled-components';

export const Wrapper = styled.section`
  display: grid;
  grid-gap: 2.4rem;
`;

export const BoardChampions = styled.div`
  display: grid;
  grid-gap: 2.4rem;
  grid-template-columns: repeat(auto-fill, minmax(20rem, 1fr));
  justify-content: space-evenly;

  @media (max-width: 856px) {
    grid-template-columns: repeat(auto-fill, minmax(15rem, 1fr));
  }

  @media (max-width: ${props => props.theme.extraSmall}) {
    justify-items: center;
  }
`;

export const Text = styled.p`
  ${props => props.theme.sans16pxItalic}
  text-align: center;
  color: rgba(255, 255, 255, 0.8);
`;
