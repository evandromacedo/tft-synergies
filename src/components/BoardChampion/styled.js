import styled from 'styled-components';
import TierBorder from '../Common/TierBorder';

export const Wrapper = styled(TierBorder)`
  position: relative;
  height: 12.8rem;
  padding: 0.8rem;
  max-width: 25.6rem;
  display: grid;
  grid-gap: 0.8rem;

  /* Background image for Splash Art */
  &:before {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    z-index: -1;
    background-image: url(${props => props.background});
    background-position: center right;
    background-size: cover;
  }
`;

export const Synergies = styled.div`
  display: grid;
  grid-gap: 0.8rem;
  grid-template-rows: repeat(3, 2rem);
`;

export const Items = styled.div`
  display: grid;
  grid-gap: 0.8rem;
  grid-template-columns: 2.8rem 2.8rem 2.8rem 1fr;
  align-items: center;
`;

export const Item = styled.div`
  width: 2.8rem;
  height: 2.8rem;
  background-color: ${props => props.theme.background3};
`;

export const Name = styled.p`
  ${props => props.theme.bold20px}
  text-align: center;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;
