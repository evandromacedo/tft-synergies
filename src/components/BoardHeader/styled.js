import styled from 'styled-components';

export const Wrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr auto;
  align-items: center;

  @media (max-width: ${props => props.theme.extraSmall}) {
    grid-template-columns: 1fr;
    grid-gap: 1.2rem;
    justify-items: center;
  }
`;
