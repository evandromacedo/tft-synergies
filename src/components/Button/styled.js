import styled from 'styled-components';

export const Button = styled.button`
  ${props => props.theme.bold16px}
  text-transform: uppercase;
  color: inherit;
  padding: 0.7rem 2rem 0.5rem 2rem;
  border-radius: 2px;
  border-top: 2px solid ${props => props.theme.accentYellow};
  border-bottom: 2px solid #67502d;
  background-image: linear-gradient(
      to bottom,
      ${props => props.theme.accentYellow},
      #67502d
    ),
    linear-gradient(${props => props.theme.accentYellow}, #67502d);
  background-size: 2px 100%;
  background-position: 0 0, 100% 0;
  background-repeat: no-repeat;

  &:hover {
    filter: brightness(130%);
  }
`;

export const Content = styled.span`
  transform: translateY(1px);
`;
