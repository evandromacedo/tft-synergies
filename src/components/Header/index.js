import React from 'react';
import Divider from '../Common/Divider';
import { HeaderWrapper, MenuWrapper, NavList } from './styled';
import Logo from '../../static/img/Logo.svg';

function Header() {
  return (
    <HeaderWrapper>
      <MenuWrapper>
        <a href="/" title="TFT Synergies Logo" data-test="logo">
          <img src={Logo} alt="Logo" />
        </a>
        <nav>
          <NavList>
            <li>
              <a href="#about">About</a>
            </li>
            <li>
              <a href="/">Contribute</a>
            </li>
          </NavList>
        </nav>
      </MenuWrapper>
      <Divider />
    </HeaderWrapper>
  );
}

export default Header;
