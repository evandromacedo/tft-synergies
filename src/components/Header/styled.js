import styled from 'styled-components';
import HeaderBackground from '../../static/img/HeaderBackground.jpg';

export const HeaderWrapper = styled.header`
  background-image: url(${HeaderBackground});
  background-position: center center;
  background-size: cover;
`;

export const MenuWrapper = styled.div`
  padding: 2.4rem 0;
  display: grid;
  grid-gap: 2.4rem;
  justify-items: center;
`;

export const NavList = styled.ul`
  width: 184px;
  display: grid;
  grid-auto-flow: column;
  justify-content: space-between;

  li > a {
    ${props => props.theme.bold20px}
    position: relative;

    &::before {
      content: ' ';
      position: absolute;
      bottom: -6px;
      height: 2px;
      width: 0;
      border-radius: 2px;
      background-color: ${props => props.theme.accentYellow};
      transition: width 0.1s linear;
    }

    &:hover::before {
      width: 100%;
    }
  }
`;
