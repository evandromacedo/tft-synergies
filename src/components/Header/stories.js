import React from 'react';

import Header from './index.js';

import { storiesOf } from '@storybook/react';

storiesOf('Header', module).add('Lol?', () => <Header />);
