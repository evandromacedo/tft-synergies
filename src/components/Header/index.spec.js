import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from '.';
import { NavList } from './styled';

describe('<Header />', () => {
  const wrapper = mount(<Header />);

  it('renders properly', () => {
    shallow(<Header />);
  });

  it('renders a header and nav tags', () => {
    expect(wrapper.exists('header')).toBe(true);
    expect(wrapper.exists('nav')).toBe(true);
  });

  it('renders the logo within a link to home', () => {
    const linkTag = wrapper.find('a[data-test="logo"]');
    expect(toJSON(linkTag)).toMatchSnapshot();
  });

  it('renders NavList having 2 items', () => {
    const NavListComponent = wrapper.find(NavList);
    const Ul = NavListComponent.find('ul');
    expect(Ul.children()).toHaveLength(2);
  });
});
