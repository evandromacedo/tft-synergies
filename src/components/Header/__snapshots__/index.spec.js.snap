// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the logo within a link to home 1`] = `
<a
  data-test="logo"
  href="/"
  title="TFT Synergies Logo"
>
  <img
    alt="Logo"
    src="Logo.svg"
  />
</a>
`;
